modules/generic_worker/manifests/init.pp

```diff
@@ -66,6 +79,72 @@
                     content => template('generic_worker/proxy-httpd.conf.erb');
             }
         }
+        Ubuntu: {
+            case $::operatingsystemrelease {
+                16.04: {
+                    if ($environment == 'staging') {
+                        # We are limited to 22 characters for worker_type
+                        $worker_type = "gecko-t-linux-talos-b"
+                        $taskcluster_client_id = secret('generic_worker_linux_staging_client_id')
+                        $taskcluster_access_token = hiera('generic_worker_linux_staging_access_token')
+                    }
+                    else {
+                        $worker_type = "gecko-t-osx-linux-talos"
+                        $taskcluster_client_id = secret('generic_worker_linux_client_id')
+                        $taskcluster_access_token = hiera('generic_worker_linux_access_token')
+                    }
+                    file {
+                        ["${::users::builder::home}/.config",
+                        "${::users::builder::home}/.config/autostart"]:
+                            ensure => directory,
+                            owner  => $users::builder::username,
+                            group  => $users::builder::group;
+                        "${::users::builder::home}/.config/autostart/gnome-terminal.desktop":
+                            content => template('generic_worker/gnome-terminal.desktop.erb'),
+                            owner   => $users::builder::username,
+                            group   => $users::builder::group;
+                    }
+                    file { '/usr/local/bin/run-generic-worker.sh':
+                        ensure  => present,
+                        content => template('generic_worker/run-generic-worker.sh.erb'),
+                        mode    => '0755',
+                        owner   => $users::root::username,
+                        group   => $users::root::group;
+                    }
+                    file { '/etc/generic-worker.config':
+                        ensure  => present,
+                        content => template('generic_worker/generic-worker.config.erb'),
+                        mode    => '0644',
+                        owner   => $users::root::username,
+                        group   => $users::root::group;
+                    }
+                    exec { 'create gpg key':
```

Comment:
This exec looks like it is already defined on line 66.  You probably don't want to duplicate that.